sdk/go/pulumi/types.go

```diff
@@ -520,7 +522,10 @@ func (o *OutputState) ApplyTWithContext(ctx context.Context, applier interface{}
 
 // IsSecret returns a bool representing the secretness of the Output
 func IsSecret(o Output) bool {
-	return o.getState().secret
+	s := o.getState()
```

Comment:
This is a .. misleading func. It doesn't await the future to complete, but secretness in the limit is only known after awaiting the result. So it can return false on a secret output at first, and then return true once it completes. Right? So it's on the caller to make sure the output is complete before calling this. I thin this remains true before and after the change. What does the lock buy?